src/senaite/sync/updatestep.py

```diff
@@ -65,7 +65,8 @@ def _fetch_data(self):
         window = 500
         overlap = 5
         effective_window = window-overlap
-        number_of_pages = (cd["count"]/effective_window) + 1
+        count = cd["count"] if 'count' in cd else 0
```

Comment:
Hi Pau. I think 'count' is not found when we receive an error message in JSON response or we don't get any response at all. So instead of setting it to '0', I think it's better to check if 'count' exists, if no, we can log it as an error and return (so last fetched time doesn't get updated in next lines).

Reply:
Thanks @nihadness ! Please, @jgallostra take note!